src/main/scala/mesosphere/mesos/ResourceMatcher.scala

```diff
@@ -51,20 +87,28 @@ object ResourceMatcher {
       badConstraints.isEmpty
     }
 
-    def portsOpt: Option[Seq[RangesResource]] = new PortsMatcher(app, offer, acceptedResourceRoles).portRanges match {
-      case None =>
-        log.warn("App ports are not available in the offer.")
-        None
+    def portsOpt: Option[Seq[RangesResource]] = new PortsMatcher(app, offer, acceptedResourceRoles).portRanges
+
+    val basicResourceRoles = Map(
+      "cpu" -> cpuRoleOpt,
+      "disk" -> diskRoleOpt,
+      "mem" -> memRoleOpt
+    )
 
-      case x @ Some(portRanges) =>
-        log.debug("Met all constraints.")
-        x
+    if (basicResourceRoles.values.exists(_.exists(!_.matches))) {
```

Comment:
Nit: is thie clearer as `! basicResourceRoles.values.forAll(_.map(_.matches).getOrElse(false))` ?